pubsub/google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -27,9 +28,40 @@
 
 
 class Dispatcher(object):
-    def __init__(self, queue, subscriber):
+    def __init__(self, manager, queue):
+        self._manager = manager
         self._queue = queue
-        self._subscriber = subscriber
+        self._thread = None
+
+    def start(self):
+        """Start a thread to dispatch requests queued up by callbacks.
+        Spawns a thread to run :meth:`dispatch_callback`.
+        """
+        if self._thread is not None:
```

Comment:
Maybe add a lock? I expect the code between lines 40 to 57 is probably more or less instant, but to be safe... :-)

Reply:
Not a bad idea, I'll add operational locks to all helper threads.